src/UCTNode.cpp

```diff
@@ -242,7 +243,11 @@ void UCTNode::accumulate_eval(float eval) {
 }
 
 UCTNode* UCTNode::uct_select_child(int color, bool is_root) {
-    LOCK(get_mutex(), lock);
+    // make sure expansion is done
```

Comment:
I wonder if this needs a check that the node can never be expanded again (i.e. against min_psa_ratio).

I guess during each search, every node only gets expanded or re-expanded once. So either we are expanding the node, or it's already expanded. But it can't get re-expanded in the middle.

Reply:
If we were to get this working then I think the right way to do this is instead of having a 'is expanding' flag, change it to an enum which is basically not expanded / expanding / expanded three-state thingy.